app/(wiki)/(articles)/obstacles/[item]/page.tsx

```diff
@@ -1,9 +1,33 @@
-import GenericMDXPageFactory, { GenericGenerateMetadataFactory, GenericGenerateStaticParamsFactory } from "@/components/layouts/GenericMDXPageFactory";
+import GenericArticlePage from "@/components/generics/GenericArticlePage";
+import ObstacleSidebar from "@/components/sidebars/ObstacleSidebar";
 import { Obstacles } from "@/vendor/suroi/common/src/definitions/obstacles";
 
-export const generateMetadata = GenericGenerateMetadataFactory(Obstacles.definitions);
-export const generateStaticParams = GenericGenerateStaticParamsFactory(Obstacles.definitions)
+const toExport = GenericArticlePage({
+  items: Obstacles.definitions,
+  path: "obstacles",
+  Sidebar: ObstacleSidebar,
+  combinedArticles: [
+    {
+      title: "Crates",
+      items: [
+        "aegis_crate",
+        "flint_crate",
+        "regular_crate",
+        "melee_crate"
+      ],
+      fileName: "crates"
+    },
+    {
+      title: "Trees",
+      items: [
+        "oak_tree",
+        "birch_tree",
+        "pine_tree"
+      ],
+      fileName: "trees"
+    }
+  ]
+});
 
-export default GenericMDXPageFactory({
-  path: "obstacles"
-})
\ No newline at end of file
+exports = toExport;
+export default toExport.default;
```

Comment:
This is an example of how the rendering refactor works.
You can specify which items get combined into one article (like all the crates, trees, etc.). Regex soon:tm:

Reply:
@kenos1 Basically it allows wiki maintainers to merge very similar articles at their discretion.

Reply:
Awesome, lgtm